### PR TITLE
add changepassword support to siac

### PR DESCRIPTION
**Pull request by DavidVorick**

### CODE REVIEW

File: siac/walletcmd.go [hunk 1]
```diff
@@ -162,6 +169,9 @@ will be sent to your wallet.`,
 
 const askPasswordText = "We need to encrypt the new data using the current wallet password, please provide: "
 
+const newPasswordText = "Please provide the current password of the wallet: "
+const oldPasswordText = "Please provide the new password for the wallet: "
+
```
Comment (lukechampine) [inline siac/walletcmd.go]: Surely "Current password" and "New password" are clear enough? Users are familiar with password rotation.

`askPasswordText` is a bit long-winded too, although that's a less familiar situation.

Comment (johnathanhowell) [inline siac/walletcmd.go]: agreed, `Current password` and `New password` are definitely enough here.

Comment (johnathanhowell) [inline siac/walletcmd.go]: Also, the wording on these is flipped.

**Comment by johnathanhowell** [comment]
latest commits fixup the prompt messages (oldpassword and newpassword had swapped semantic meanings), fixes an unintended branch in `walletChangePasswordHandler`, and improves TestWalletChangePasswordDeep, verifying that passwords with spaces work correctly.

**Pull request merged**